src/test/java/io/cdap/plugin/gcp/gcs/sink/GCSBatchSinkCmekKeyTest.java

```diff
@@ -0,0 +1,155 @@
+    FieldSetter
+      .setField(gcsBatchSinkConfig, GCSBatchSink.GCSBatchSinkConfig.class.getDeclaredField("path"), path);
+    FieldSetter.setField(gcsBatchSinkConfig, GCSBatchSink.GCSBatchSinkConfig.class.getDeclaredField("format")
+      , "csv");
+    FieldSetter
+      .setField(gcsBatchSinkConfig, GCPReferenceSinkConfig.class.getDeclaredField("referenceName")
+        , "testref");
+    FieldSetter
+      .setField(gcsBatchSinkConfig, GCPConfig.class.getDeclaredField("serviceAccountType"), serviceAccountType);
+    if (!Strings.isNullOrEmpty(serviceFilePath)) {
+      FieldSetter
+        .setField(gcsBatchSinkConfig, GCPConfig.class.getDeclaredField("serviceFilePath"), serviceFilePath);
+    }
+    if (!Strings.isNullOrEmpty(serviceAccountJson)) {
+      FieldSetter
+        .setField(gcsBatchSinkConfig, GCPConfig.class.getDeclaredField("serviceAccountJson"), serviceAccountJson);
+    }
+    return gcsBatchSinkConfig;
+  }
+
+  @Test
+  public void testServiceAccountPath() throws Exception {
+    GCSBatchSink.GCSBatchSinkConfig config = getConfig(project, bucketPath, "filePath",
+                                                       serviceAccountFilePath, null);
+    testValidCmekKey(config);
+    testInvalidCmekKey(config);
+  }
+
+  @Test
+  public void testServiceAccountJson() throws Exception {
+    GCSBatchSink.GCSBatchSinkConfig config = getConfig(project, bucketPath, "JSON",
+                                                       null, serviceAccountKey);
+    testValidCmekKey(config);
+    testInvalidCmekKey(config);
+  }
+
+  public void testValidCmekKey(GCSBatchSink.GCSBatchSinkConfig config) throws Exception {
+    MockFailureCollector collector = new MockFailureCollector("gcssink");
+    config.validate(collector);
+    FieldSetter
+      .setField(config, GCSBatchSink.GCSBatchSinkConfig.class.getDeclaredField("location"), "us-east1");
+    FieldSetter
+      .setField(config, GCSBatchSink.GCSBatchSinkConfig.class.getDeclaredField("cmekKey"),
+                "projects/cdf-test-322418/locations/us-east1/keyRings/my_ring/cryptoKeys/test_key");
+    config.validate(collector);
+    Assert.assertEquals(0, collector.getValidationFailures().size());
+  }
+
+  public void testInvalidCmekKey(GCSBatchSink.GCSBatchSinkConfig config) throws Exception {
+    MockFailureCollector collector = new MockFailureCollector("gcssink");
+    config.validate(collector);
+    FieldSetter
+      .setField(config, GCSBatchSink.GCSBatchSinkConfig.class.getDeclaredField("cmekKey"),
+                "keyRings/my_ring/cryptoKeys/test_key");
+    config.validate(collector);
+    FieldSetter
+      .setField(config, GCSBatchSink.GCSBatchSinkConfig.class.getDeclaredField("location"), "us");
+    FieldSetter
+      .setField(config, GCSBatchSink.GCSBatchSinkConfig.class.getDeclaredField("cmekKey"),
+                "projects/cdf-test-322418/locations/us-east1/keyRings/my_ring/cryptoKeys/test_key");
+    config.validate(collector);
```

Comment:
tests should be as focused as possible. Make the validateCmekKey package private and call it directly.